### PR TITLE
feat: callbacks is configured from callbacks not from callback groups

### PR DESCRIPTION
**Pull request by h-suzuki-isp**
## Description

The architecture's callbacks were created from callback_groups, so the specification has been changed to create them from callback_loaded.

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] (Optional) The PR has been properly tested with CARET_report verification.
- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

**Review by ymski — CHANGES_REQUESTED** [review]

File: src/caret_analyze/architecture/struct/node.py [hunk 1]
```diff
@@ -80,9 +80,13 @@ def timers(self) -> list[TimerStruct]:
 
     @property
     def callbacks(self) -> list[CallbackStruct] | None:
-        if self._callback_groups is None:
+        service_callbacks = [service.callback for service in self.services]
```
[the remark this answers is not in the document]
Comment (h-suzuki-isp) [reply src/caret_analyze/architecture/struct/node.py:83]: I understand. Fixed in https://github.com/tier4/caret_analyze/pull/506/commits/08dd93cfe84801d398f590493039c40729e69e50.

[the remark this answers is not in the document]
Comment (h-suzuki-isp) [reply src/caret_analyze/architecture/struct/node.py]: Fixed in https://github.com/tier4/caret_analyze/pull/506/commits/d4bd08c56609f30e14a51d51a46e5828ea043c05.

File: src/test/architecture/test_architecture_loader.py [hunk 2]
```diff
@@ -418,7 +418,7 @@ def assigned(node_paths, message_contexts):
         assert node.timers == [timer]
         assert node.subscriptions == [subscription]
         assert node.services == [service]
-        assert node.callbacks == [callback]
+        assert node.callbacks == [service.callback, subscription.callback, timer.callback]
```
Comment (ymski) [inline src/test/architecture/test_architecture_loader.py:421]: `service.callback` is Mock.
It must be configured to return service_callback in advance as follows.
```
service_callback = mocker.Mock(spec=ServiceCallbackStruct)
subscription_callback = mocker.Mock(spec=SubscriptionCallbackStruct)
timer_callback = mocker.Mock(spec=TimerCallbackStruct)
mocker.patch.object(subscription, 'callback', service_callback)
mocker.patch.object(service, 'callback', subscription_callback)
mocker.patch.object(timer, 'callback', timer_callback)
```

**Review by ymski — CHANGES_REQUESTED** [review]

File: src/caret_analyze/architecture/architecture.py [hunk 3]
```diff
@@ -535,9 +540,11 @@ def rename_callback(self, src: str, dst: str) -> None:
...
 
         """
-        cb_s: list[CallbackStruct] =\
-            Util.flatten(cb_g.callbacks for cb_g in
-                         Util.flatten([e.callback_groups for e in self._executors]))
+        cb_s: list[CallbackStruct] = []
+        for node in self._nodes:
+            if node.callbacks is None:
+                continue
+            cb_s += [callback for callback in node.callbacks]
```
[the remark this answers is not in the document]
Comment (h-suzuki-isp) [reply src/caret_analyze/architecture/architecture.py]: Fixed in https://github.com/tier4/caret_analyze/pull/506/commits/4399eab04f7542ee1c1a89376d77eded3ee8c144.

File: src/test/architecture/test_architecture_loader.py (hunk 2 above)
Comment (h-suzuki-isp) [reply src/test/architecture/test_architecture_loader.py:428]: Fixed in https://github.com/tier4/caret_analyze/pull/506/commits/5a9871e13ca3a5ee646d814c9abe37187cad6526.

**Comment by h-suzuki-isp** [comment]
This PR will be handed over to https://github.com/tier4/caret_analyze/pull/511.

**Pull request closed without merging**